api/environments/identities/traits/models.py

```diff
@@ -57,6 +57,13 @@ def natural_key(self):
     def trait_value(self):
         return self.get_trait_value()
 
+    @property
+    def transient(self) -> bool:
+        try:
+            return self._is_transient
+        except AttributeError:
+            return self.pk is None
```

Comment:
Possible not an issue, but I'm a _little_ concerned that this will bleed into other logic. A trait that just hasn't been saved yet isn't really _transient_, right? But it would be according to this logic. 

Maybe this should actually only be defined as a `SerializerMethodField`? (and maybe `_is_transient` should be a public attribute?)

Reply:
I've reimplemented the property to be more cleaner and decoupled it from model state.

Reply:
Nice use of a `setter` 👌